language-design.md

```diff
@@ -28,7 +41,9 @@ you know a poorly designed language? What are the symptoms?
 
 **Response**
 
+Several features that identify a well-designed language are the following: easy to extend, comprehensible, usability, good documentation, non-bitter user base. Essentially, a well designed language won’t leave its users frustrated for hours on end as they struggle to program the basic ideas they have. In other words, since a programming language is a way of expression, a good language should be intuitive and self consistent. It will have documentation to support its design and the syntax will be readable (enough). Furthermore, as Steele highlights often in _Growing a Language_, it will have the ability to be extended and thus adapted by its user base. This evolution allows users to grow the language in a direction that best suits the needs of the masses. An example of this, as explored by Mernik, shows how the COM interface allows Excel’s limited macro language to interface with languages such as C++, Java, and Basic through a “process of componentization… called automation [Chappell 1996]” [Mernik et al., 2005, 318]. By allowing a language, while staying specific to its domain, be more accessible to other interfaces, its use cases grow. This engenders a positive user base that speaks well of the language, rather than the stink eye from the computer science community.
 
+Conversely, a poorly designed language will feature the opposite of basically everything described above. Its documentation will be unclear, its syntax unintuitive, its design unextendable, and its user base very aggravated. Furthermore, it may lack implementation for different tasks a user may need. That being said, the symptoms of a of a poorly designed language can also be less obvious. For instance, as discussed by Steele, APL, a language the author himself admittedly loves, died primarily because “there was no way for a user to grow the language in a smooth way” [Steele, 1998, 6]. Thus, a poorly designed language can suffer from a range of shortcoming, but it takes only one major Achilles Heel to kill an otherwise good language.
```

Comment:
Your last insight is intriguing - I wonder to what degree user bases of PLs are willing to deal with single glaring problems. I can't really think of other languages that have died out to look at (mostly because, surprisingly, they will have died out) but I'm curious how many languages have died just because users weren't willing to deal with one problem, or if in those cases that problem was really fixable or if it was a more fundamental mistake in the design of the language. It could even be that some major problems were intended as features that just weren't suited to what the user base wound up consisting of.

Overall a lot of the symptoms you've mentioned are similar to ours - documentation, syntax, etc. I definitely agree that those are good signs within the language itself, but I really like your idea that you can look at the users to tell if the language is good or not.